hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/hub/OfflineScanner.java

```diff
@@ -70,28 +76,43 @@ boolean offlineScan(final DetectProject detectProject, final HubScanConfig hubSc
-        CLILocation cliLocation = new CLILocation(intLogger, hubScanConfig.getToolsDir());
-        if (StringUtils.isNotBlank(hubSignatureScannerOfflineLocalPath)) {
-            cliLocation = new OfflineCLILocation(intLogger, new File(hubSignatureScannerOfflineLocalPath));
-        }
-
-        boolean cliInstalledOkay = checkCliInstall(cliLocation, new SilentLogger());
-        if (!cliInstalledOkay && StringUtils.isNotBlank(detectConfiguration.getHubSignatureScannerHostUrl())) {
-            installSignatureScannerFromUrl(intLogger, hubScanConfig);
-            cliInstalledOkay = checkCliInstall(cliLocation, intLogger);
-        }
+        final ProjectRequestBuilder projectRequestBuilder = new ProjectRequestBuilder();
+        projectRequestBuilder.setProjectName(detectProject.getProjectName());
+        projectRequestBuilder.setVersionName(detectProject.getProjectVersion());
 
-        if (!cliInstalledOkay && StringUtils.isNotBlank(hubSignatureScannerOfflineLocalPath)) {
-            logger.warn(String.format("The signature scanner is not correctly installed at %s", hubSignatureScannerOfflineLocalPath));
-            return false;
-        } else if (!cliInstalledOkay) {
-            logger.warn(String.format("The signature scanner is not correctly installed at %s", hubScanConfig.getToolsDir()));
-            return false;
-        } else {
-            simpleScanUtility.setupAndExecuteScan(cliLocation);
-            logger.info(String.format("The scan dry run files can be found in : %s", simpleScanUtility.getDataDirectory()));
-            return true;
+        final ExecutorService executorService = Executors.newFixedThreadPool(detectConfiguration.getHubSignatureScannerParallelProcessors());
+        try {
+            final ParallelSimpleScanner parallelSimpleScanner = new ParallelSimpleScanner(intLogger, intEnvironmentVariables, gson, executorService);
+            CLILocation cliLocation = new CLILocation(intLogger, hubScanConfig.getCommonScanConfig().getToolsDir());
+            if (StringUtils.isNotBlank(hubSignatureScannerOfflineLocalPath)) {
+                cliLocation = new OfflineCLILocation(intLogger, new File(hubSignatureScannerOfflineLocalPath));
+            }
+
+            boolean cliInstalledOkay = checkCliInstall(cliLocation, new SilentLogger());
+            if (!cliInstalledOkay && StringUtils.isNotBlank(detectConfiguration.getHubSignatureScannerHostUrl())) {
+                installSignatureScannerFromUrl(intLogger, hubScanConfig);
+                cliInstalledOkay = checkCliInstall(cliLocation, intLogger);
+            }
+
+            if (!cliInstalledOkay && StringUtils.isNotBlank(hubSignatureScannerOfflineLocalPath)) {
+                logger.warn(String.format("The signature scanner is not correctly installed at %s", hubSignatureScannerOfflineLocalPath));
+                return Collections.emptyList();
```

Comment:
Not sure I agree with the multiple return paths here, especially in an if/else-if/else block. Instead, I recommend scanTargetOutputs to be created outside the if and populated within it, then returned before the finally block.